### ISSUE TITLE
Regional images / bosses

### ISSUE THREAD

**Issue by swistak** [julianrupp/tswlairmgr#1]
Heya. Great job on the app first of all. Second of all would it be possible for you to add regional images?

**Comment by julianrupp** [comment]
Thanks!

Regional fragments actually is something that's already on my feature to do list [on the official forums](https://forums.thesecretworld.com/showthread.php?84221-Webapp-TSW-Lair-Manager&p=1837689#post1837689).

There is no ETA however, and the way I want to integrate it will probably require a bigger refactor, so it's going to be a rather big chunk of work. Already got the graphics and fragment data (which lair boss drops which regional fragment(s)).

Edit: Whoops, actually it wasn't even on the list in the forum thread. Has been on my personal list though :)

**Comment by swistak** [comment]
If you can upload images I can take a look at the code and hope you incorporate it into the app? :)

**Comment by julianrupp** [comment]
The images are already in the [resource repository](https://github.com/julianrupp/tswlairmgr-resources):
- [Aspect of the Long-Toothed](https://github.com/julianrupp/tswlairmgr-resources/blob/master/graphics/lairbosses/bosses_xray_all_wtransparent/sol.png)
- [Aspect of the Many-Limbed](https://github.com/julianrupp/tswlairmgr-resources/blob/master/graphics/lairbosses/bosses_xray_all_wtransparent/egy.png)
- [Aspect of the Great-Winged](https://github.com/julianrupp/tswlairmgr-resources/blob/master/graphics/lairbosses/bosses_xray_all_wtransparent/tra.png)

However, I'd really prefer to do this feature myself, as I already have a quite detailed vision of how I want it to work and look like in my head.

Maybe I can postpone the refactor one more time for this... might get to check that in the next few days.

**Comment by julianrupp** [comment]
Just a quick update.

A few days ago I began [rewriting the whole thing](https://github.com/julianrupp/tswlairmgr/tree/rewrite) as I came to the conclusion that adding a new feature this big would be too much of a hassle with the old code base.

I've now finished the groundwork – module system, lair data, localization support facilities, and a shared MVC component for displaying lair boss and regional boss fragments. Basically, everything but the actual functionality is there.

I'll first re-add the core functionality that was already present in the old code base (lair fragment organizer / pick table) in one module, and after that I'll create another module that will provide fragment lookup functionality:
- Select a fragment from a dropdown menu, and display it within its associated fragment set (= Boss), and
- select a boss from a dropdown menu and display its fragment set.
- For regional boss fragments, there will be some way (hover / click) to display a list of the lair bosses that have a chance to drop the respective regional fragment, and clicking them looks up that boss.
- I'll probably include a lookup history to provide a "back to previously looked up item" function.

At least that's my vision currently.

So. This is going to be done. Regionals are coming :)
